columnphysics/icepack_shortwave.F90

```diff
@@ -4815,16 +4813,33 @@ subroutine compute_dEdd_5bd(                          &
                   ks = ext_cff_mss_ice_drc(ns,nmbrad_snicar)
                   ws = ss_alb_ice_drc     (ns,nmbrad_snicar)
                   gs = asm_prm_ice_drc    (ns,nmbrad_snicar)
-               elseif (ceiling(rsnw(ksnow)) - rsnw(ksnow) < 1.0e-3_dbl_kind) then
-                  ! radius = 30 --> nr = 1 in SNICAR table      ! NOTE
-                  nr = ceiling(rsnw(ksnow)) - 30 + 1            ! hardwired min radius = 30
-                  ks = ext_cff_mss_ice_drc(ns,nr)
-                  ws = ss_alb_ice_drc     (ns,nr)
-                  gs = asm_prm_ice_drc    (ns,nr)
-               else ! linear interpolation in rsnw
-                  nr = ceiling(rsnw(ksnow)) - 30 + 1            ! hardwired min radius = 30
-                  delr = (rsnw(ksnow) - floor(rsnw(ksnow))) &   ! hardwired delta radius = 1 in table
-                       / (ceiling(rsnw(ksnow)) - floor(rsnw(ksnow))) ! denom always = 1
+               else
+                  ! check whether rsnw_snicar_chk are integers with delta of one, makes search faster
+                  if (rsnw_snicar_chk == 'unknown') then
+                     rsnw_snicar_chk = 'delta1'
+                     do n = 1,size(rsnw_snicar_tab)
+                        if (abs(rsnw_snicar_tab(n) - real(nint(rsnw_snicar_tab(n)),kind=dbl_kind)) > 1.0e-6) then
+                           rsnw_snicar_chk = 'mixed'
+                        endif
+                     enddo
+                     do n = 2,size(rsnw_snicar_tab)
+                        if (abs(rsnw_snicar_tab(n) - rsnw_snicar_tab(n-1) - c1) > 1.0e-6) then
```

Comment:
why do you need 2 loops, one with n=1,size() using real(nint()) and the other with n=2,size() without real(nint()) but with -1?

Reply:
These loops are checking too different conditions.  The first is that the array is basically all integers (to roundoff-ish).  The second is that the delta between two elements in the array is 1.0 (to roundoff-ish).  Both of those need to be true for the "shortcut" to work.

Reply:
ok thx